### ISSUE TITLE
Button tooltip

### ISSUE THREAD

**Issue by konghu** [carbon-design-system/carbon#3667]
Button React component provides a default tooltip with default description set. I wonder if this is necessary. When I use the button components without providing icon description as props, the default description doesn't really make sense in terms of a11y.

<img width="223" alt="Screen Shot 2019-08-07 at 9 32 43 AM" src="https://user-images.githubusercontent.com/21697034/62626871-5344e380-b8f6-11e9-971c-df7574620ff5.png">

## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?

> What happened instead?

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by elizabethsjudd** [comment]
Buttons with text shouldn't need additional descriptions on the icon, the icon can use `aria-hidden="true"` in this use case.

**Comment by konghu** [comment]
@elizabethsjudd Great thanks, that's exactly what I am looking for

**Comment by konghu** [comment]
Actually I see that by default the SVG comes with `aria-hidden="true"` attribute, but that still doesn't get rid of tooltip

<img width="445" alt="Screen Shot 2019-08-07 at 10 07 51 AM" src="https://user-images.githubusercontent.com/21697034/62629500-3ced5680-b8fb-11e9-84ab-db3ce941cd0c.png">

**Comment by elizabethsjudd** [comment]
When you use `aria-hidden="true"` you should not need the `aria-label` or `alt` attributes on the icon or the `title` element within the icon.  Semantically they contradict each other.

**Comment by konghu** [comment]
Yes, that is right, but the aria-label is supplied from the Button component's default props. I didn't want to add additional description

`
Button.defaultProps = {
  iconDescription: 'Provide icon description if icon is used',
  tabIndex: 0,
  type: 'button',
  disabled: false,
  small: false,
  kind: 'primary',
};
`

**Comment by dakahn** [comment]
Looks like we're properly hiding buttons with labels _and_ icons as seen here
https://react.carbondesignsystem.com/?path=/story/button--playground
